components/movementsensor/imuvectornav/imu.go

```diff
@@ -24,12 +24,15 @@ import (
 
 var model = resource.DefaultModelFamily.WithModel("imu-vectornav")
 
+// defaultBaudRate is the value we assign if there is no value specified in the attributes.
+const defaultBaudRate = 9600
+
 // Config is used for converting a vectornav IMU MovementSensor config attributes.
 type Config struct {
-	SPI   string `json:"spi_bus"`
-	Speed *int   `json:"spi_baud_rate"`
-	Pfreq *int   `json:"polling_freq_hz"`
-	CSPin string `json:"chip_select_pin"`
+	SPI      string `json:"spi_bus"`
+	BaudRate *int   `json:"spi_baud_rate,omitempty"`
```

Comment:
@susmitaSanyal @oliviamiller @randhid 

is there a specific reason for why `BaudRate` and `Pfreq` are pointers to an int?

I am asking since if I choose to leave them as empty in the attributes section of the config, the constructor freaks out. 
However, if I change them to be of type `int` the problem goes away.

Reply:
It was a trick to check if a config field was present in the validator back in the day when we didn't have good json schemas for the configs. We can make them not pointers for now to avoid nil checks.

Reply:
hooray!